src/main/java/com/puppycrawl/tools/checkstyle/filters/PatchXpathFilterElement.java

```diff
@@ -175,6 +186,25 @@ private boolean lineMatching(int currentLine) {
         return result;
     }
 
+    private boolean lineMatching(int childAstStartLine, int childAstEndLine) {
+        boolean result = false;
+        for (List<Integer> singleLineRangeList : lineRangeList) {
+            final int startLine = singleLineRangeList.get(0) + 1;
+            final int endLine = singleLineRangeList.get(1) + 1;
+            if (startLine == endLine) {
+                result = childAstStartLine <= startLine && startLine <= childAstEndLine;
```

Comment:
please just increment endLine here to use common logic for all cases below.

Reply:
@baratali sorry, I am not clear.

Reply:
I mean this:
```java
            if (startLine == endLine) {
                result = childAstStartLine <= startLine && startLine <= childAstEndLine;
            }
            else {
                result = (childAstStartLine <= startLine && startLine <= childAstEndLine)
                        || (childAstStartLine <= endLine - 1 && endLine - 1 <= childAstEndLine);
            }
```
--->
```java
            if (startLine == endLine) {
                endLine++;
            }

            result = (childAstStartLine <= startLine && startLine <= childAstEndLine)
                    || (childAstStartLine <= endLine - 1 && endLine - 1 <= childAstEndLine);
```

Reply:
ok, done